alerting/src/main/kotlin/org/opensearch/alerting/settings/AlertingSettings.kt

```diff
@@ -156,5 +173,94 @@ class AlertingSettings {
             -1L,
             Setting.Property.NodeScope, Setting.Property.Dynamic
         )
+
+        val TOTAL_MAX_ACTIONS_PER_TRIGGERS = Setting.intSetting(
+            "plugins.alerting.max_actions_across_triggers",
+            DEFAULT_TOTAL_MAX_ACTIONS_PER_TRIGGERS,
+            -1, MaxActionsTriggersValidator(internalClient),
+            Setting.Property.NodeScope, Setting.Property.Dynamic
+        )
+        val TOTAL_MAX_ACTIONS_ACROSS_TRIGGERS = Setting.intSetting(
+            "plugins.alerting.total_max_actions_across_triggers",
+            DEFAULT_TOTAL_MAX_ACTIONS_ACROSS_TRIGGERS,
+            -1, TotalMaxActionsTriggersValidator(internalClient),
+            Setting.Property.NodeScope, Setting.Property.Dynamic
+        )
+
+        internal class TotalMaxActionsTriggersValidator(val client: Client?) : Setting.Validator<Int> {
+            override fun validate(value: Int) {}
+
+            override fun validate(value: Int, settings: Map<Setting<*>, Any>) {
+                val maxActions = settings[TOTAL_MAX_ACTIONS_PER_TRIGGERS] as Int
+                validateActionsTrigger(maxActions, value, client)
+            }
+
+            override fun settings(): MutableIterator<Setting<*>> {
+                val settings = mutableListOf<Setting<*>>(
+                    TOTAL_MAX_ACTIONS_PER_TRIGGERS
+                )
+                return settings.iterator()
+            }
+        }
+
+        internal class MaxActionsTriggersValidator(val client: Client?) : Setting.Validator<Int> {
+            override fun validate(value: Int) {}
+
+            override fun validate(value: Int, settings: Map<Setting<*>, Any>) {
+                val totalMaxActions = settings[TOTAL_MAX_ACTIONS_ACROSS_TRIGGERS] as Int
+                validateActionsTrigger(value, totalMaxActions, client)
+            }
+
+            override fun settings(): MutableIterator<Setting<*>> {
+                val settings = mutableListOf<Setting<*>>(
+                    TOTAL_MAX_ACTIONS_ACROSS_TRIGGERS
+                )
+                return settings.iterator()
+            }
+        }
+
+        private fun validateActionsTrigger(maxActions: Int, totalMaxActions: Int, client: Client?) {
```

Comment:
Where do we validate `maxActions`? Should we not validate the number of actions there are per trigger in each monitor?